src/hotspot/cpu/x86/methodHandles_x86.cpp

```diff
@@ -122,17 +122,63 @@ void MethodHandles::verify_ref_kind(MacroAssembler* _masm, int ref_kind, Registe
   __ bind(L);
 }
 
-#endif //ASSERT
+void MethodHandles::verify_method(MacroAssembler* _masm, Register method, Register temp, vmIntrinsics::ID iid) {
+  BLOCK_COMMENT("verify_method {");
+  __ verify_method_ptr(method);
+  if (VerifyMethodHandles) {
+    Label L_ok;
+
+    const Register method_holder = temp;
+    __ load_method_holder(method_holder, method);
+    __ push(method_holder); // keep holder around for diagnostic purposes
+
+    switch (iid) {
+      case vmIntrinsicID::_invokeBasic:
+        // Require compiled LambdaForm class to be fully initialized.
+        __ cmpb(Address(method_holder, InstanceKlass::init_state_offset()), InstanceKlass::fully_initialized);
+        __ jccb(Assembler::equal, L_ok);
```

Comment:
Should this also be long jump?

Reply:
It jumps over the code generated by `MacroAssembler::stop`, so a short jump instruction should be fine here.

I did the opposite: turned the jump for other linkers from `jcc` to `jccb`.